### PR TITLE
Use GPU picking for line(like) primitives, fix `interactive` flags

### LINKED ISSUE #1021: Hover over image under point / rect as well
When hovering over an image with points or rectangles on it, the image hover preview suddenly disappears when we hover over a point or rectangle. This is particularly annoying because the position of the point or rectangle boundary is usually the most interesting place to inspect the image.

hover points on image

### PR DESCRIPTION
**Pull request by Wumpf**
Almost done with #1615, only depth-clouds remaining for what we planned out.

Plan was to use gpu picking for all lines/arrows/boxes/etc., everything that is done via the line renderer, but hit things on the way so it got a bit more than that:

* Add picking ids to line renderer
    * line renderer takes now line builder directly
* Fix many instances of not using `interactive` property correctly, by moving this check to the recently centralized picking method, i.e. solving this as a "postprocess-step"
* simplify picking code a lot using the fact that there's 3 kinds of picking now (gpu, ui overlay, image rect). Fixes #1021
* switch to the established batch processing iteration on arrow (scene part)

Image hover info always there and stable at bottom of selection:

https://user-images.githubusercontent.com/1220815/231700405-438c68ca-6c82-42fb-9016-f54a23a50390.mov

Still a little bit of flickering since the hover things randomly decide to flip upwards - I think this happens on some of the first frames in a row where a point is picked as well. @emilk ?

Screenshot demonstrating radius awareness of picking (even taking into account the arrow head)
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

### CODE REVIEW

File: crates/re_viewer/src/ui/view_spatial/scene/picking.rs [hunk 1]
```diff
@@ -421,36 +248,35 @@ fn picking_textured_rects(
         let v = dir_from_rect_top_left.dot(rect.extent_v) / rect.extent_v.length_squared();
 
         if (0.0..=1.0).contains(&u) && (0.0..=1.0).contains(&v) {
-            let picking_hit = PickingRayHit {
+            hits.push(PickingRayHit {
                 instance_path_hash: *id,
-                ray_t: t,
-                info: AdditionalPickingInfo::TexturedRect(glam::vec2(u, v)),
+                space_position: context.ray_in_world.origin + context.ray_in_world.dir * t,
```
Comment (emilk) [inline crates/re_viewer/src/ui/view_spatial/scene/picking.rs:253]:
```suggestion
                space_position: context.ray_in_world.point_along(t),
```

**Comment by emilk** [comment]
> Image hover info always there and stable at bottom of selection:
https://user-images.githubusercontent.com/1220815/231700405-438c68ca-6c82-42fb-9016-f54a23a50390.mov
Still a little bit of flickering since the hover things randomly decide to flip upwards - I think this happens on some of the first frames in a row where a point is picked as well. @emilk ?

This is indeed quite annoying. Please open an issue about it. This is probably an immediate mode thing - egui should probably hide the tooltips the very first frame, because it stores their sizes and uses that to properly position them.

**Comment by Wumpf** [comment]
>  Is it to keep all process_entity_view similar-ish?

Yeah, followed a pre-existing pattern there, but sure why not let's do cleanup there. Traitification any day now ;D

File: crates/re_viewer/src/ui/view_spatial/scene/picking.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_viewer/src/ui/view_spatial/scene/picking.rs:253]: neat

**Pull request merged**